chain/neutrino.go

```diff
@@ -358,17 +359,30 @@ func (s *NeutrinoClient) Rescan(startHash *chainhash.Hash, addrs []btcutil.Addre
 	// notification indicating the rescan has "finished".
 	if header.BlockHash() == *startHash {
 		s.finished = true
+		rescanQuit := s.rescanQuit
+
+		// Release the lock while dispatching the notification since
+		// it's possible for the notificationHandler to be waiting to
+		// acquire it before receiving the notification.
+		s.clientMtx.Unlock()
 		select {
 		case s.enqueueNotification <- &RescanFinished{
 			Hash:   startHash,
 			Height: int32(bestBlock.Height),
 			Time:   header.Timestamp,
 		}:
 		case <-s.quit:
+			// We'll need to re-acquire the lock in order for the
+			// deferred release call to not panic.
+			s.clientMtx.Lock()
 			return nil
-		case <-s.rescanQuit:
+		case <-rescanQuit:
+			// We'll need to re-acquire the lock in order for the
+			// deferred release call to not panic.
+			s.clientMtx.Lock()
```

Comment:
Could we instead get rid of the defer and unlock manually on each return case?

Reply:
@cfromknecht seems to prefer the defer approach, though that might've been because there were some fields that were being used without the lock in the prior diff.

Reply:
Yeah I’m fine with not having a defer, originally thought it’d make a smaller diff but seems not :P

Reply:
Done.